### PR TITLE
Fix format_to + FMT_STRING for wide character type

### PR DESCRIPTION
**Pull request by hmbj**
Fixes issue #3925

### CODE REVIEW

**Comment by hmbj** [comment]
Not sure if this is the optimal fix, though :)

**Review by vitaut — CHANGES_REQUESTED** [review]
Thanks for the PR. I think it's better to keep `detail::to_string_view` and instead add a `format_to` overload that takes `wformat_string` similar to this one: https://github.com/fmtlib/fmt/blob/aa52eb765dec0eb67fab4b035a4a05b82346c8f5/include/fmt/xchar.h#L140

File: include/fmt/xchar.h [hunk 1]
```diff
@@ -127,6 +127,18 @@ auto join(const std::tuple<T...>& tuple, basic_string_view<wchar_t> sep)
   return {tuple, sep};
 }
 
+template <typename... T>
+auto format(wformat_string<T...> fmt, T&&... args) -> std::wstring {
+  return vformat(fmt::wstring_view(fmt), fmt::make_wformat_args(args...));
+}
```
Comment (vitaut) [inline include/fmt/xchar.h:133]: There is no need to move this function definition, please revert.

Comment (hmbj) [reply include/fmt/xchar.h]: I've reverted the move.

This patch only fixes format_to called with a compile time string (FMT_STRING). Looks like the other functions calling detail::to_string_view in this file has the same problem, like `vformat_to`, `vformat` and maybe `formatted_size`...

**Review by vitaut — APPROVED** [review]

File: include/fmt/xchar.h (hunk 1 above)
Comment (vitaut) [reply include/fmt/xchar.h]: `vformat*` are fine because they take runtime format strings.

**Comment by vitaut** [comment]
Merged, thanks.

**Pull request merged**